includes/Classifai/Providers/Azure/Personalizer.php

```diff
@@ -0,0 +1,608 @@
+			return $content;
+		}
+
+		// phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+		$post_id   = $content->rewardActionId;
+		$event_id  = $content->eventId; // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+		$post      = get_post( $post_id );
+		$post_link = esc_url( get_permalink( $post ) );
+		$title     = get_the_title( $post );
+
+		if ( ! $title ) {
+			$title = __( '(no title)', 'classifai' );
+		}
+
+		$markup = '<li>';
+
+		if ( $attributes['displayFeaturedImage'] && has_post_thumbnail( $post ) ) {
+			$image_classes = 'wp-block-classifai-recommended-content__featured-image';
+
+			$featured_image = get_the_post_thumbnail( $post );
+			if ( $attributes['addLinkToFeaturedImage'] ) {
+				$featured_image = sprintf(
+					'<a href="%1$s" aria-label="%2$s" class="classifai-send-reward" data-eventid="%3$s">%4$s</a>',
+					esc_url( $post_link ),
+					esc_attr( $title ),
+					esc_attr( $event_id ),
+					$featured_image
+				);
+			}
+			$markup .= sprintf(
+				'<div class="%1$s">%2$s</div>',
+				esc_attr( $image_classes ),
+				$featured_image
+			);
+		}
+
+		$markup .= sprintf(
+			'<a href="%1$s" class="classifai-send-reward" data-eventid="%2$s">%3$s</a>',
+			esc_url( $post_link ),
+			esc_attr( $event_id ),
+			esc_html( $title )
+		);
+
+		if ( isset( $attributes['displayAuthor'] ) && $attributes['displayAuthor'] ) {
+			$author_display_name = get_the_author_meta( 'display_name', $post->post_author );
+
+			/* translators: byline. %s: current author. */
+			$byline = sprintf( __( 'by %s', 'classifai' ), $author_display_name );
+
+			if ( ! empty( $author_display_name ) ) {
+				$markup .= sprintf(
+					'<div class="wp-block-classifai-recommended-content__post-author">%1$s</div>',
+					esc_html( $byline )
+				);
+			}
+		}
+
+		if ( isset( $attributes['displayPostDate'] ) && $attributes['displayPostDate'] ) {
+			$markup .= sprintf(
+				'<time datetime="%1$s" class="wp-block-classifai-recommended-content__post-date">%2$s</time>',
+				esc_attr( get_the_date( 'c', $post ) ),
+				esc_html( get_the_date( '', $post ) )
+			);
+		}
+
+		if ( isset( $attributes['displayPostExcept'] ) && $attributes['displayPostExcept'] ) {
+			$trimmed_excerpt = get_the_excerpt( $post );
+
+			if ( post_password_required( $post ) ) {
+				$trimmed_excerpt = __( 'This content is password protected.', 'classifai' );
+			}
+
+			$markup .= sprintf(
+				'<div class="wp-block-classifai-recommended-content__post-excerpt">%1$s</div>',
+				esc_html( $trimmed_excerpt )
+			);
+		}
+
+		$markup .= "</li>\n";
+
+		$class = 'wp-block-classifai-recommended-content';
+
+		if ( isset( $attributes['displayPostDate'] ) && $attributes['displayPostDate'] ) {
+			$class .= ' has-dates';
+		}
+
+		if ( isset( $attributes['displayAuthor'] ) && $attributes['displayAuthor'] ) {
+			$class .= ' has-author';
+		}
+
+		$wrapper_attributes = get_block_wrapper_attributes( array( 'class' => $class ) );
```

Comment:
A couple things here:

1. When testing things locally, I'm getting PHP warnings originating from here: `Trying to access array offset on value of type null in "wp-includes/class-wp-block-supports.php" lines 94 and 96`
2. This function was added in WP 5.6.0 which is above our minimum requirements. If we need to use this, we'll need to bump our minimums to 5.6

Reply:
We are not using this now. however, we already updated wp minimum to 5.6 in other PR.